### PR TITLE
Build: Make distroless image default and publish mimir-alpine

### PR DESCRIPTION
**Pull request by ying-jeanne**
#### Which issue(s) this PR fixes or relates to

Make distroless the default base image for grafana/mimir. To minimize the change, i just switched dockerfiles and changed makefile, pipeline stayed mostly untouched.

#### Checklist

- n/a Tests updated.
- n/a Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- n/a [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

**Comment by ying-jeanne** [comment]
wait for this ticket to be implemented https://grafana.freshservice.com/support/tickets/32446

**Comment by ying-jeanne** [comment]
the run for testing also alpine during deploy https://github.com/grafana/mimir/actions/runs/9275402810

**Comment by ying-jeanne** [comment]
From ci, except deploy failed since mimir-alpine repo doesn't exist yet, PR works as expected

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 ### Grafana Mimir
 
+* [CHANGE] Build: Publish distroless as grafana/mimir image by default. #8204
```
Comment (pstibrany) [inline CHANGELOG.md:7]:
```suggestion
* [CHANGE] Build: `grafana/mimir` docker image is now based on `gcr.io/distroless/static-debian12` image. Alpine-based docker image is still available as `grafana/mimir-alpine`, until Mimir 2.15. #8204
```

(This is our plan, right?)

File: Makefile [hunk 2]
```diff
@@ -126,32 +126,40 @@ SED ?= $(shell which gsed 2>/dev/null || which sed)
...
 			--build-arg=BINARY_SUFFIX=_race \
-			--build-arg=BASEIMG="gcr.io/distroless/base-nossl-debian12" \
-			-t $(IMAGE_PREFIX)$(shell basename $(@D))-distroless:$(IMAGE_TAG_RACE) $(@D)/; \
+			--build-arg=EXTRA_PACKAGES="gcompat" \
+			-t $(IMAGE_PREFIX)$(shell basename $(@D))-alpine:$(IMAGE_TAG_RACE) $(@D)/; \
 	fi;
 	@echo
 	# We need gcompat -- compatibility layer with glibc, as race-detector currently requires glibc, but Alpine uses musl libc instead.
-	$(SUDO) docker build --build-arg=revision=$(GIT_REVISION) --build-arg=goproxyValue=$(GOPROXY_VALUE) --build-arg=USE_BINARY_SUFFIX=true --build-arg=BINARY_SUFFIX=_race --build-arg=EXTRA_PACKAGES="gcompat" -t $(IMAGE_PREFIX)$(shell basename $(@D)):$(IMAGE_TAG_RACE) $(@D)/
+	# for grafana/mimir which is based on distroless image, the EXTRA_PACKAGES is just ignored.
```
Comment (pstibrany) [inline Makefile:141]: Why do we even pass `EXTRA_PACKAGES` to the build then? We can remove it.

**Review by pstibrany — APPROVED** [review]
Looks good, left minor comments. Thank you.

File: Makefile (hunk 2 above)
Comment (ying-jeanne) [reply Makefile:141]: since this is used in other dockerfile still, not only for cmd/mimir, we still need it.

Comment (ying-jeanne) [reply Makefile:141]: I see what you mean now,  it is UPTODATE_RACE target which is only for cmd/mimir, removing it.

File: Makefile [hunk 3]
```diff
@@ -126,32 +126,38 @@ SED ?= $(shell which gsed 2>/dev/null || which sed)
...
 			--build-arg=goproxyValue=$(GOPROXY_VALUE) \
 			--build-arg=USE_BINARY_SUFFIX=true \
 			--build-arg=BINARY_SUFFIX=_race \
-			--build-arg=BASEIMG="gcr.io/distroless/base-nossl-debian12" \
-			-t $(IMAGE_PREFIX)$(shell basename $(@D))-distroless:$(IMAGE_TAG_RACE) $(@D)/; \
+			--build-arg=EXTRA_PACKAGES="gcompat" \
+			-t $(IMAGE_PREFIX)$(shell basename $(@D))-alpine:$(IMAGE_TAG_RACE) $(@D)/; \
 	fi;
 	@echo
 	# We need gcompat -- compatibility layer with glibc, as race-detector currently requires glibc, but Alpine uses musl libc instead.
```
Comment (pstibrany) [inline Makefile:140]: This comment should be moved to "alpine" version. For distroless image, we should restore comment about using distroless/base-nossl-debian12 as base image.

**Pull request merged**